Assets/Scripts/FireBallStats.cs

```diff
@@ -8,36 +8,70 @@ public class FireBallStats : ProjectileWeapon
     public GameObject fireBall;
     GameObject newFireBall;
     public Transform firePoint;
+    private Vector2 spawn;
 
     public float launchForce;
     public float recoil;
     public float upForce;
-    public float chargeTimer;
+    private float chargeTimer;
+    private float posNeg;
+    private bool rocket;
+    public int damage;
 
-    private float damage;
     private float FireBallSize = 3f;
+    public GameObject player;
+    
 
-    public override void Fire()
+    public override void Fire(Vector2 direction)
     {
-        newFireBall = Instantiate(fireBall, firePoint.position, firePoint.rotation);
+        damage = damage / 10;
+        player = GameObject.Find("TempPlayer");
+        spawn = player.GetComponent<Transform>().position;
+     
+        spawn.x += posNeg;
+        if (rocket)
+        {
+            spawn.y -= 2;
+            spawn.x = player.GetComponent<Transform>().position.x;
+            
+        }
+        fireBall.GetComponent<Transform>().position = spawn;
+        newFireBall = Instantiate(fireBall, fireBall.transform.position, fireBall.transform.rotation);
         FireBallSize += chargeTimer;
-
+        if(rocket) {
+            newFireBall.GetComponent<Rigidbody2D>().velocity = (newFireBall.transform.up * -1 * launchForce);
+            player.GetComponent<Rigidbody2D>().velocity = (player.transform.up * 10);
```

Comment:
For the player please do set the velocity directly. Instead use `Rigidbody.AddForce(direction * intensity * mass, ForceMode2D.Inpulse)` It's almost the same, but it doesn't remove any velocity the player already has.

Reply:
I see that you actually do something similar on like 48, however the mass part is important because that affects the amount of force you want to apply on something

Reply:
ALSO, you probably didn't know this but for specific reasons like these I actually added a function in the PlatformerController called KnockBack. You can call it like this `PlatformerController.instance.KnockBack(direction, intensity)` this also helps reduce the number of GetComponent calls

Reply:
instead of you getting knocked back when you shoot. the actual explosion should be what knocks you back/up. Like a rocket jump. You can use [overlapcircle](https://docs.unity3d.com/ScriptReference/Physics2D.OverlapCircle.html) to find all colliders in an area of the explosion. and for each collider with a rigid body, you would AddForce away from the center of the explosion. Let me or Jacob know if you have questions